gamemodes/terrortown/entities/weapons/weapon_tttbasegrenade.lua

```diff
@@ -64,74 +66,210 @@ CreateConVar("ttt_nade_throw_during_prep", "0", {FCVAR_ARCHIVE, FCVAR_NOTIFY})
 function SWEP:SetupDataTables()
 	self:NetworkVar("Bool", 0, "Pin")
 	self:NetworkVar("Int", 0, "ThrowTime")
+	self:NetworkVar("Bool", 1, "ShotToActivate")
+	self:NetworkVar("Float", 0, "AnimTimer")
 end
 
 ---
 -- @ignore
 function SWEP:PrimaryAttack()
 	self:SetNextPrimaryFire(CurTime() + self.Primary.Delay)
-
-	if GetRoundState() == ROUND_PREP and not GetConVar("ttt_nade_throw_during_prep"):GetBool() then
-		return
-	end
-
+	if GetRoundState() == ROUND_PREP and not GetConVar("ttt_nade_throw_during_prep"):GetBool() then return end
 	self:PullPin()
 end
 
 ---
 -- @ignore
 function SWEP:SecondaryAttack()
+end
 
+---
+-- @param Vector P The initial position for this iteration cycle.
+-- @param number V The launch velocity for this iteration cycle.
+-- @param number G The gravity multiplier for this iteration cycle.
+-- @param number T The step size for the current iteration cycle.
+-- @return Vector The next location to iterate from.
+local function PositionFromPhysicsParams(P, V, G, T)
+	local A = G * physenv.GetGravity()
+
+	return P + (V * T + 0.5 * A * T ^ 2)
+end
+
+if CLIENT then
+	---
+	-- @realm client
+	local fuse_ui = CreateConVar("ttt2_grenade_fuse_meter_ui", 0, FCVAR_ARCHIVE)
+
+	---
+	-- @realm client
+	local trajectory_ui = CreateConVar("ttt2_grenade_trajectory_ui", 0, FCVAR_ARCHIVE)
+
+	local function AlphaLerp(from, frac)
+		local fr = frac ^ (1 / 2)
```

Comment:
```suggestion
		local fr = frac ^ 0.5
```
No reason to put an extra division here